File: RT/Programming/I5/TP2/dessin.rb
require "java"

class Dessin<java.awt.Canvas

  # =========== PROPERTIES ==========



  # =========== CONSTRUCTOR ==========

  def initialize

  end



  # =========== METHODS ==========

  def paint(g)
    g.setXORMode(java.awt.Color.white)

    for i in 0..20 do
      g.fillRect(100 + i*10, 100, 100, 100)
      sleep(0.3)
      g.fillRect(100 + i*10, 100, 100, 100)
    end

  end



  # =========== GET & SET ==========



  # ================================

end




require "java"

import java.awt.Container;
import javax.swing.JButton;
import javax.swing.JFrame;
import javax.swing.JLabel;

@width = 500;
@height = 300;
@posX = 900;
@posY = 500;

dessin = Dessin.new();
dessin.setPreferredSize(java.awt.Dimension.new(500,300)) # /!\

frame = JFrame.new("Window 1");
c = frame.getContentPane();

c.add(dessin);

frame.setLocationRelativeTo(nil);
frame.setSize(@width, @height);
frame.setVisible(true);
frame.setDefaultCloseOperation(JFrame::EXIT_ON_CLOSE);
frame.setLocation(@posX, @posY);

label = JLabel.new("Bonjour !");
c.add(label);

b1 = JButton.new("Test");
b1.setSize(100, 20);
b1.setLocation(75, 5);
label.add(b1);

#moveWindow(frame)

=begin
#def moveWindow(f)
  centerX = frame.getX() + frame.getWidth() / 2;
  centerY = frame.getY() + frame.getHeight() / 2;
  rayon = 30;

  while (true)
    for i in 0..(2 * Math::PI)
      i = i + Math::PI / 12.0
      sleep(0.100)
      frame.setLocation((centerX + rayon * Math.sin(i)).to_i(), (centerY + 30 * Math.cos(i)).to_i);
      frame.setVisible(true);
    end
  end
#end
=end

File: RT/Programming/I5/TP2_Java/src/fr/vincent/tp2/Dessin.java
package fr.vincent.tp2;

import java.awt.Canvas;
import java.awt.Color;
import java.awt.Graphics;
import java.awt.Polygon;
import java.util.ArrayList;
import java.util.List;

import javax.swing.JFrame;

public class Dessin extends Canvas
{
	// PROPERTIES



	// CONSTRUCTOR

	public Dessin()
	{
		super();
	}



	// AUTO-CALL

	public void paint(Graphics g)
	{
		//drawMovingRect(g);
		//drawAll(g);

		drawObstacles(g);
		drawRect(g);
	}



	// METHODS

	public void drawObstacles(Graphics g)
	{
		g.setColor(Color.BLUE);

		for (Rect iter : Core._obstacles)
		{
			g.fillRect((int)iter.getX(), (int)iter.getY(), 5, 5);
		}

		g.setColor(Color.RED);
	}

	public void drawRect(Graphics g)
	{
		g.fillRect((int)Core._rect.getX(), (int)Core._rect.getY(), (int)Core._rect.getWidth(), (int)Core._rect.getHeight());
	}

	public void drawMovingRect(Graphics g)
	{
		g.setXORMode(java.awt.Color.white);

		for (int i = 0; i < 20; i++) 
		{
			g.fillRect(100 + i*10, 100, 100, 100);
			try	{ Thread.sleep(300); }
			catch (InterruptedException e) { e.printStackTrace(); }
			g.fillRect(100 + i*10, 100, 100, 100);
		}
	}

	public void drawAll(Graphics g)
	{
		g.setColor(Color.GRAY);
		g.drawRect(10, 10, 50, 20);

		g.setColor(Color.BLUE);
		g.fillRect(70, 10, 40, 22);

		g.setColor(Color.MAGENTA);
		g.drawLine(100, 5, 260, 25);

		g.setColor(Color.GREEN);
		g.drawRoundRect(10, 60, 60, 20, 3, 3);

		g.setColor(Color.BLACK);
		g.drawString("rect", 12, 12);

		g.setColor(Color.ORANGE);
		g.drawOval(10, 120, 50, 20);

		g.setColor(Color.RED);
		Polygon polygon = new Polygon();
		polygon.addPoint(15, 200);
		polygon.addPoint(42, 221);
		polygon.addPoint(38, 249);
		polygon.addPoint(54, 215);
		polygon.addPoint(15, 200);
		/*polygon.addPoint(15, 300);
		polygon.addPoint(15, 300);
		polygon.addPoint(15, 300);*/
		g.drawPolygon(polygon);
	}



}
